DOCSP-40099: Add note about client.connect() (#894)
mongodb/docs-node: source/fundamentals/connection/connect.txt

.. literalinclude:: /code-snippets/connection/stable-api-connect.js
   :language: javascript

.. note::

   As of {+driver-short+} v4.7, the driver automatically calls the ``MongoClient.connect()``
   method when using the client to perform CRUD operations on your MongoDB deployment.
   Call the ``MongoClient.connect()`` method explicitly to verify that the connection is
   successful.

> 

Author: @nbbeeken in the #node-driver-docs Slack channel posted proof that this automatic connection still works with bulk ops.

"Call the ``MongoClient.connect()`` method explicitly _if you would like to_ verify that the connection is successful" gets flagged by Vale for the italicized portion.

Reviewer: How about: "...if you want to manually verify that the connection..."? As worded, it sounds like we're directing users to do it, which isn't necessary.

.. literalinclude:: /code-snippets/connection/stable-api-connect.js
   :language: javascript

.. note::

>    As of v4.7, the {+driver-short+} will automatically call the ``MongoClient.connect()``

Reviewer: S: specify driver version (as opposed to server version)

```suggestion
   As of {+driver-short+} v4.7, the driver will automatically call the ``MongoClient.connect()``
```

>    As of v4.7, the {+driver-short+} will automatically call the ``MongoClient.connect()``

Reviewer: I: present tense

```suggestion
   As of v4.7, the {+driver-short+} automatically calls the ``MongoClient.connect()``
```

>    method when using the client to interact with your MongoDB instance, **except** when

Reviewer: I: the prerequisite of "using the client to interact with..." isn't clear to me. what does the user have to do to get the driver to call this automatically?

>    method when using the client to interact with your MongoDB instance, **except** when

Reviewer: I: there was a whole thing about a year ago about server vs. deployment vs. instance. i think the upshot was:

- MongoDB Server = the product
- MongoDB deployment = a running instance of MongoDB
- MongoDB instance = not preferred

```suggestion
   method when using the client to interact with your MongoDB deployment, **except** when
```

>    method when using the client to interact with your MongoDB instance, **except** when

Reviewer: S: the sentence is a little long, and the "except" clause seems pretty important. better as its own sentence?

>    you use the client to call bulk operations. We recommend manually calling
>    ``MongoClient.connect()`` if you want granular control and insight over
>    your application's connections.

Reviewer: S: i think it would be useful to be more specific here; "granular control and insight" is correct, but might not be helpful to a newer user. (you could also add info about bulk writes here)

```suggestion
   you use the client to call bulk operations. Call the ``MongoClient.connect()`` method
   explicitly if you are performing bulk operations or if you want to verify that the
   connection was successful.
```

.. note::

>    As of {+driver-short+} v4.7, the driver automatically calls the ``MongoClient.connect()``

Reviewer: S: this is presumably getting backported to all active docs versions, and v4.7 is in legacy. Do we need to say when this behavior started?

```suggestion
   The {+driver-short+} automatically calls the ``MongoClient.connect()``
```

Author: Well spotted, happy to remove.
